Src/BootCamp.Chapter/Player.cs

```diff
@@ -28,34 +34,55 @@ public class Player
         /// <summary>
         /// Player items. There can be multiple of items with same name.
         /// </summary>
-        private Inventory _inventory;
+        private PlayerInventory _inventory;
         /// <summary>
         /// Needed only for the extra task.
         /// </summary>
         private Equipment _equipment;
 
-        public Player()
+        public Player(string name)
         {
+            _name = name;
+            _inventory = new PlayerInventory();
+            _equipment = new Equipment();
+            _hp = baseHP;
+            _carryWeight = baseCarryWeight;
+            _strenght = 0;
+        }
+
+        public string GetName()
+        {
+            return _name;
         }
 
         /// <summary>
         /// Gets all items from player's inventory
         /// </summary>
-        public Item[] GetItems()
+        public void ShowInventory()
         {
-            return new Item[0];
+            _inventory.ShowAllItems();
+          
         }
 
         /// <summary>
         /// Adds item to player's inventory
         /// </summary>
+        // The inventory method 'AddItem' will throw an expection if no room is available in the inventory for an another item
         public void AddItem(Item item)
         {
+            try
```

Comment:
It's not that excpetional that inventory is full. Instead, check the inventory size before adding more items to it. throwing an exception is costly and it should also be reserved for unexpected situations.